src/platform/packages/shared/kbn-esql-validation-autocomplete/src/shared/helpers.ts

```diff
@@ -113,30 +111,6 @@ export function isIncompleteItem(arg: ESQLAstItem): boolean {
 export const within = (position: number, location: ESQLLocation | undefined) =>
   Boolean(location && location.min <= position && location.max >= position);
 
-function isMathFunction(query: string) {
-  const queryTrimmed = query.trimEnd();
-  // try to get the full operation token (e.g. "+", "in", "like", etc...) but it requires the token
-  // to be spaced out from a field/function (e.g. "field + ") so it is subject to issues
-  const [opString] = queryTrimmed.split(' ').reverse();
-  // compare last char for all math functions
-  // limit only to 2 chars operators
-  const fns = operatorsDefinitions.filter(({ name }) => name.length < 3).map(({ name }) => name);
-  const tokenMatch = fns.some((op) => opString === op);
-  // there's a match, that's good
-  if (tokenMatch) {
-    return true;
-  }
-  // either there's no match or it is the case where field/function and op are not spaced out
-  // e.g "field+" or "fn()+"
-  // so try to extract the last char and compare it with the single char math functions
-  const singleCharFns = fns.filter((name) => name.length === 1);
-  return singleCharFns.some((c) => c === opString[opString.length - 1]);
-}
```

Comment:
This was too automatic for me. It's not a bad idea to set things up to rely on the definitions, but not all operators require the editor marker after them.

Reply:
Agreed